iml-gui/crate/src/components/breadcrumbs.rs

```diff
@@ -0,0 +1,108 @@
+use crate::{generated::css_classes::C, Msg, Page};
+use seed::{prelude::*, *};
+use std::cmp::PartialEq;
+use std::collections::LinkedList;
+
+pub struct BreadCrumbs<Crumb> {
+    crumbs: LinkedList<Crumb>,
+}
+
+// XXX Automatic deriving would require Crumb to have
+// XXX the Default implementation, but do not use it.
+impl<Crumb> Default for BreadCrumbs<Crumb> {
+    fn default() -> Self {
+        Self {
+            crumbs: LinkedList::new(),
+        }
+    }
+}
+
+impl<Crumb: PartialEq> BreadCrumbs<Crumb> {
+    pub fn iter(self: &Self) -> impl DoubleEndedIterator<Item = &Crumb> {
+        self.crumbs.iter()
+    }
+
+    pub fn clear(self: &mut Self) -> &mut Self {
+        self.crumbs.clear();
+        self
+    }
+
+    pub fn push(self: &mut Self, n: Crumb) -> &mut Self {
+        let mut new_crumbs = LinkedList::new();
+
+        while let Some(c) = self.crumbs.pop_front() {
+            if c != n {
+                new_crumbs.push_back(c);
+            } else {
+                break;
+            }
+        }
+
+        new_crumbs.push_back(n);
+        self.crumbs = new_crumbs;
+        self
+    }
+}
+
+pub fn view(bc: &BreadCrumbs<Page>) -> impl View<Msg> {
+    let mut ol = ol![class![C.justify_center, C.truncate, C.mx_4, C.rtl]];
+    let mut it = bc.iter().rev().enumerate(); // XXX rev() because of "direction: rtl;"
+
+    while let Some((n, p)) = it.next() {
```

Comment:
nit: this is a bit more succinct by just consuming the enumerated iterator directly:

`for (n, p) in bc.iter().enumerate() {`